### DISCUSSION TITLE
Nested FSM possibility

### DISCUSSION THREAD

**Discussion by Zhuikov** [aiogram/aiogram#1184] [Q&A]
Hello! I've got a question about nested Finite State Machine possibility.

For example, there are 2 FSMs:
* `FSM_1`:
    * starts by `/fsm1`
    * states: `FSM_1_1` and `FSM_1_2`
    * cancels by `/cancel`
* `FSM_2`:
    * starts by `/fsm2`
    * states: `FSM_2_1` and `FSM_2_2`
    * cancels by `/cancel` (too)

I want to implement following behavior.

1. User sends `/fsm1`
2. User is currently in `FSM_1_1`
3. User sends `/fsm2`
4. User is currently in `FSM_2_1`
5. User sends `/cancel`
6. _User is returning to `FSM_1_1`_

Now aiogram sets state to `None` after `/cancel` command, but I want to set state from previous `FSM_1`. Similarly, I want to return to the previous `FSM_1` state, if user completes the entire `FSM_2`.

Are there any examples or best practices how to implement nested FSM? I use `aiogram==2.19`

**Comment by JrooTJunior** [comment]
It seems like you don't understand how finite state machine actually works, so, please read the docs about FSM https://docs.aiogram.dev/en/dev-3.x/dispatcher/finite_state_machine/index.html

This page is for 3.0 version but basic theories is the same as for 2.x version.
